src/System.Text.Json/tests/Serialization/CyclicTests.cs

```diff
@@ -17,5 +17,14 @@ public static void WriteCyclicFail()
             // We don't allow cycles; we throw InvalidOperation instead of an unrecoverable StackOverflow.
             Assert.Throws<InvalidOperationException>(() => JsonSerializer.ToString(obj));
         }
+
+        [Fact]
+        public static void WriteTestClassWithArrayOfElementsOfTheSameClassWithoutCyclesDoesNotFail()
+        {
+            TestClassWithArrayOfElementsOfTheSameClass obj = new TestClassWithArrayOfElementsOfTheSameClass();
+
+            //It shouldn't throw when there is no real cycle reference, and just empty object is created
+            Assert.Null(Record.Exception(() => JsonSerializer.ToString(obj)));
```

Comment:
I believe the xunit [ActiveIssue] attribute is used in cases like this.

Reply:
Yes, please update the test to reflect what **should** happen, and then add `[ActiveIssue]` to skip it for now.

Reply:
@steveharter @ahsonkhan updated accordingly to suggestions.